crates/dataflow-jit/src/codegen/intrinsics.rs

```diff
@@ -1336,6 +1337,24 @@ unsafe extern "C" fn parse_bool_from_str(
     }
 }
 
+unsafe extern "C" fn parse_decimal_from_str(
+    ptr: *const u8,
+    len: usize,
+    output: &mut MaybeUninit<u128>,
+) -> bool {
+    let string = unsafe { str_from_raw_parts(ptr, len) };
+    match string.parse::<Decimal>() {
```

Comment:
https://docs.feldera.io/docs/sql/decimal: Casting a string to a decimal value will produce the value 0 when parsing fails.
In general this behavior depends on the actual data type, sometimes is should be NULL.

Reply:
Yah, in the future I'll add the ability to react to parse failures but it depends on adts

Reply:
Until then I will file an issue about this after we merge this PR.

Reply:
I believe there's already one regarding this